components/formats-gpl/src/loci/formats/in/ImprovisionTiffReader.java

```diff
@@ -111,28 +108,38 @@ public void close(boolean fileOnly) throws IOException {
     return noPixels ? null : files;
   }
 
-  /* @see loci.formats.IFormatReader#get8BitLookupTable() */
+  /* @see loci.formats.IFormatReader#get8BitLookupTable(int) */
   @Override
-  public byte[][] get8BitLookupTable() throws FormatException, IOException {
+  public byte[][] get8BitLookupTable(int no) throws FormatException, IOException {
     FormatTools.assertId(currentId, true, 1);
-    if (readers == null || lastFile < 0 || lastFile >= readers.length ||
+
+    int[] zct = getZCTCoords(no);
+    int lastFile = FormatTools.getIndex("XYZCT", getSizeZ(), getEffectiveSizeC(),
+      getSizeT(), getImageCount(), zct[0], zct[1], zct[2]) % files.length;
+
+    if (readers == null || lastFile >= readers.length ||
       readers[lastFile] == null)
     {
-      return super.get8BitLookupTable();
+      return super.get8BitLookupTable(0);
     }
-    return readers[lastFile].get8BitLookupTable();
+    return readers[lastFile].get8BitLookupTable(0);
   }
 
-  /* @see loci.formats.IFormatReader#get16BitLookupTable() */
+  /* @see loci.formats.IFormatReader#get16BitLookupTable(int) */
   @Override
-  public short[][] get16BitLookupTable() throws FormatException, IOException {
+  public short[][] get16BitLookupTable(int no) throws FormatException, IOException {
     FormatTools.assertId(currentId, true, 1);
-    if (readers == null || lastFile < 0 || lastFile >= readers.length ||
+
+    int[] zct = getZCTCoords(no);
+    int lastFile = FormatTools.getIndex("XYZCT", getSizeZ(), getEffectiveSizeC(),
+      getSizeT(), getImageCount(), zct[0], zct[1], zct[2]) % files.length;
```

Comment:
Actually, no, it' can't be moved since lastFile is used in both cases.  But I don't spot any problems--both files and readers are initialised in initStandardMetadata, and the LookupTable methods aren't valid to call until after this is completed.  I might be missing something subtle, or not so subtle, though.